apps/server/src/modules/deletion/client/deletion.client.ts

```diff
@@ -22,36 +25,73 @@ export class DeletionClient {
 
 		// Prepare the POST /deletionRequests endpoint beforehand to not do it on every client call.
 		this.postDeletionRequestsEndpoint = new URL('/admin/api/v1/deletionRequests', this.baseUrl).toString();
+		this.postDeletionExecutionsEndpoint = new URL('/admin/api/v1/deletionExecutions', this.baseUrl).toString();
 	}
 
 	async queueDeletionRequest(input: DeletionRequestInput): Promise<DeletionRequestOutput> {
-		const request = this.httpService.post(this.postDeletionRequestsEndpoint, input, this.defaultHeaders());
-
-		return firstValueFrom(request)
-			.then((resp: AxiosResponse<DeletionRequestOutput>) => {
-				// Throw an error if any other status code (other than expected "202 Accepted" is returned).
-				if (resp.status !== 202) {
-					throw new Error(`invalid HTTP status code in a response from the server - ${resp.status} instead of 202`);
-				}
-
-				// Throw an error if server didn't return a requestId in a response (and it is
-				// required as it gives client the reference to the created deletion request).
-				if (!resp.data.requestId) {
-					throw new Error('no valid requestId returned from the server');
-				}
-
-				// Throw an error if server didn't return a deletionPlannedAt timestamp so the user
-				// will not be aware after which date the deletion request's execution will begin.
-				if (!resp.data.deletionPlannedAt) {
-					throw new Error('no valid deletionPlannedAt returned from the server');
-				}
-
-				return resp.data;
-			})
-			.catch((err: Error) => {
-				// Throw an error if sending/processing deletion request by the client failed in any way.
-				throw new Error(`failed to send/process a deletion request: ${err.toString()}`);
-			});
+		let resp: AxiosResponse<DeletionRequestOutput>;
+
+		try {
+			const request = this.httpService.post(this.postDeletionRequestsEndpoint, input, this.defaultHeaders());
+
+			resp = (await firstValueFrom(request)) as AxiosResponse<DeletionRequestOutput>;
+		} catch (err: unknown) {
+			// Throw an error if sending deletion request has failed.
+			throw new BadGatewayException('DeletionClient:queueDeletionRequest', ErrorUtils.createHttpExceptionOptions(err));
+		}
+
+		// Throw an error if any other status code (other than expected "202 Accepted" is returned).
+		if (resp.status !== 202) {
```

Comment:
You are sure that i ist correctly? It can't 200 been?
Maybe check if status >= 300 ???

Reply:
Yes it is correct. The response should be exactly HTTP 202 (Accepted) as this is the proper status code for the operation the Admin API will be performing here through the methods exposed from the deletion module - it just queue the deletion, not performing it synchronously or in any immediate manner - the final deletion is triggered by the second method (in another process). More info on the HTTP 202 and its meaning: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202 .